### PR TITLE
Fix some glitches

### PR DESCRIPTION
**Pull request by anaelChardan**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,33 +15,33 @@ When persisting this collection of values (Mysql, Elasticsearch), we use intensi
 
 ## Problems
 
-Do not that in theory, these problems should be highlited by the use of the profiler.
+Do note that, in theory these problems should be highlited by the use of the profiler.
 It's the role of the profile to determine if an optimization worths an improvement or not.
 
-Here, we are doing the opposite: it is to better understand the problem in the Blackfire profile.
+Here, we are doing the opposite: it is better to understand the problem in a Blackfire profile.
```
Comment (ahocquard) [inline README.md]: no, "it is to better understand" is the real signification

File: README.md [hunk 2]
```diff
@@ -15,33 +15,33 @@ When persisting this collection of values (Mysql, Elasticsearch), we use intensi
...
 The more you have normalizers in the registry, slower it is to find the good normalizer.
 
-So, when normalizing value collection of 1000 products values and having 15 normalizers registered in the serializer, a lot of time is wasted at finding the normalizer for the product values. 
+So, when normalizing a value collection of 1000 products values and having 15 normalizers registered in the serializer, a lot of time is wasted at finding the normalizer for the product values. 
 
-Note: The complexity is O(m * n) where "m" is the number of product values and "n" the number of normalizers declared in the registry.
+Note: The complexity  of looping on it is O(m * n) where "m" is the number of product values and "n" the number of normalizers declared in the registry.
 
-The solution to avoid this complexity is to call directly the normalizer instead of calling the serializer.
+The solution to avoid this complexity is to call directly the normalizer instead of calling the serializer, or to have a map with supports function (only for class) and Normalizer themselves which is O(1).
```
Comment (ahocquard) [inline README.md]: Actually, it's useless to have a map if you can inject it directly. I prefer to keep it as before.

File: README.md [hunk 3]
```diff
@@ -15,33 +15,33 @@ When persisting this collection of values (Mysql, Elasticsearch), we use intensi
...
 
 Each value is merged into the normalized value collection by using `array_merge_recursive`.
 The problem is pretty obvious when you know how this function works.
 
 The `$result` variable has one more item per iteration in the loop.
-As the arrays to be merged are passed by copy, it copies a bigger and bigger array.
+As the arrays to be merged are passed by copy, it copies an array which becomes bigger and bigger.
 
-Note: we can easily proove that it's a O(n^2) complexity as the complexity is a suite 1 + 2 + 3 + ... + "m", where "m" is the number of product values.
+Note: we can easily proove that it's a O(n^2) complexity as the complexity is a suite 1 + 2 + 3 + ... + "n", where "n" is the number of product values.
```
Comment (ahocquard) [inline README.md]: no real rule, you know ;) it was to keep the consistency with "### Normalizer pattern"

Comment (anaelChardan) [reply README.md]: So you have to change the complexity in O(m^2) as it is the limit.

File: README.md (hunk 2 above)
Comment (anaelChardan) [reply README.md]: I do agree ;), I just wanted to add more explanation

File: README.md (hunk 1 above)
Comment (anaelChardan) [reply README.md]: Quite weird, we have to find a way to reformulate it.

Comment (ahocquard) [reply README.md]: https://www.linguee.fr/anglais-francais/traduction/to+better+understand.html

File: README.md (hunk 2 above)
Comment (ahocquard) [reply README.md]: Yes, but it's already very verbose.

File: README.md (hunk 3 above)
Comment (ahocquard) [reply README.md]: Ah, I didn't see that. Ok keep it like it.

File: README.md (hunk 1 above)
Comment (anaelChardan) [reply README.md]: Ok :D

**Review by ahocquard — APPROVED** [review]

**Comment by ahocquard** [comment]
Thanks ❤️

**Comment by anaelChardan** [comment]
<3

**Pull request merged**